substrate/frame/support/src/lib.rs

```diff
@@ -2262,6 +2262,221 @@ pub mod pallet_macros {
+	/// be written as `<T: Config>`, in the generated code, it is always `<T: Config>`.
+	///
+	/// #### Example:
+	/// 
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     # use frame_support::pallet_prelude::*;
+	///     # #[pallet::config]
+	///     # pub trait Config: frame_system::Config {}
+	///     # #[pallet::pallet]
+	///     # pub struct Pallet<T>(_);
+	/// 	/// Unnamed syntax, without bounding `T`.
+	///     #[pallet::storage]
+	///     pub type Foo<T> = StorageValue<_, u32>;
+	///
+	/// 	/// Unnamed syntax, with bounding `T`.
+	///     #[pallet::storage]
+	///     pub type Bar<T: Config> = StorageValue<_, u32>;
+	///
+	/// 	/// Named syntax.
+	///     #[pallet::storage]
+	///     pub type Baz<T> = StorageMap<_, Hasher = Blake2_128Concat, Key = u32, Value = u32>;
+	/// }
+	/// ```
+	///
+	/// ### Query Type
+	///
+	/// Every storage type mentioned above has a generic type called `QueryKind` that determines its "query" type. 
+	/// This refers to the kind of value returned when querying the storage, for instance, through a `::get()` method. 
+	///
+	/// There are three types of queries:
+	///
+	/// 1. [`OptionQuery`](frame_support::storage::types::OptionQuery): The default query type. It returns `Some(V)` if the value is present, or 
+	/// `None` if it isn't, where `V` is the value type.
+	/// 2. [`ValueQuery`](frame_support::storage::types::ValueQuery): Returns the value itself if present; otherwise, it returns 
+	/// `Default::default()`. This behavior can be adjusted with the `OnEmpty` generic parameter, which defaults to `OnEmpty = GetDefault`.
+	/// 3. [`ResultQuery`](frame_support::storage::types::ResultQuery): Returns `Result<V, E>`, where `V` is the value type, and `E` is the pallet error type defined by the [`#[pallet::error]`](frame_support::pallet_macros::error) attribute.
+	///
+	/// ### Appending
+	///
+	/// All storage items — such as `StorageValue`, `StorageMap`, and their variants—offer an `::append()` method 
+	/// optimized for collections. Using this method avoids the inefficiency of decoding and re-encoding 
+	/// entire collections when adding items. For instance, consider the storage declaration 
+	/// `type MyVal<T> = StorageValue<_, Vec<u8>, ValueQuery>`. With `MyVal` storing a large list of bytes, 
+	/// `::append()` lets you directly add bytes to the end in storage without processing the full list. 
+	/// Depending on the storage type, additional key specifications may be needed.
+	///
+	/// ### Optimized Length Decoding
+	///
+	/// All storage items — such as `StorageValue`, `StorageMap`, and their counterparts — 
+	/// incorporate the `::decode_len()` method. This method allows for efficient retrieval 
+	/// of a collection's length without the necessity of decoding the entire dataset.
+	///
+	/// ### Hashers
+	///
+	/// For all storage types, except `StorageValue`, a set of hashers needs to be specified. The choice
+	/// of hashers is crucial, especially in production chains. The purpose of
```

Comment:
This is one that I would probably tackle in a ref-doc, as it is too detailed. Good for now I would say.